tools/export/sw4stm32/__init__.py

```diff
@@ -14,95 +14,374 @@
+            'name': 'NUCLEO-F446ZE',
+            'mcuId': 'STM32F446ZETx'
+        },
+        'NUCLEO_F746ZG':
+        {
+            'name': 'NUCLEO-F746ZG',
+            'mcuId': 'STM32F746ZGTx'
+        },
+        'NUCLEO_F767ZI':
+        {
+            'name': 'NUCLEO-F767ZI',
+            'mcuId': 'STM32F767ZITx'
+        },
+        'NUCLEO_L011K4':
+        {
+            'name': 'NUCLEO-L011K4',
+            'mcuId': 'STM32L011K4Tx'
+        },
+        'NUCLEO_L031K6':
+        {
+            'name': 'NUCLEO-L031K6',
+            'mcuId': 'STM32L031K6Tx'
+        },
+        'NUCLEO_L053R8':
+        {
+            'name': 'NUCLEO-L053R8',
+            'mcuId': 'STM32L053R8Tx'
+        },
+        'NUCLEO_L073RZ':
+        {
+            'name': 'NUCLEO-L073RZ',
+            'mcuId': 'STM32L073RZTx'
+        },
+        'NUCLEO_L152RE':
+        {
+            'name': 'NUCLEO-L152RE',
+            'mcuId': 'STM32L152RETx'
+        },
+        'NUCLEO_L432KC':
+        {
+            'name': 'NUCLEO-L432KC',
+            'mcuId': 'STM32L432KCUx'
+        },
+        'NUCLEO_L476RG':
+        {
+            'name': 'NUCLEO-L476RG',
+            'mcuId': 'STM32L476RGTx'
+        },
     }
 
     TARGETS = BOARDS.keys()
 
-    def __gen_dir(self, dirname):
-        settings = join(self.export_dir, dirname)
+    def __gen_dir(self, dir_name):
+        """
+        Method that creates directory
+        """
+        settings = join(self.export_dir, dir_name)
         mkdir(settings)
 
     def __generate_uid(self):
+        """
+        Method to create random int
+        """
         return "%0.9u" % randint(0, 999999999)
 
+    @staticmethod
+    def filter_dot(path):
+        """
+        This function removes ./ from str.
+        str must be converted with win_to_unix() before using this function.
+        """
+        if not path:
+            return path
+        if path.startswith('./'):
+            return path[2:]
+        return path
+
+    def build_excludelist(self):
+        """
+        This method creates list for excluded directories.
+        """
+        self.source_folders = [self.filter_dot(s)
+                               for s in set(dirname(src) for src in
+                                            self.resources.c_sources +
+                                            self.resources.cpp_sources +
+                                            self.resources.s_sources)]
+        if '.' in self.source_folders:
+            self.source_folders.remove('.')
+
+        top_folders = [f for f in set(s.split('/')[0]
+                                      for s in self.source_folders)]
+
+        for top_folder in top_folders:
+            for root, dirs, files in walk(top_folder, topdown=True):
+                # Paths returned by os.walk() must be split with os.dep
+                # to accomodate Windows weirdness.
+                parts = root.split(sep)
+                self.remove_unused('/'.join(parts))
```

Comment:
When I looked at the code first time, my thought was, that there should be some common or base package every exporter (or more likely every exporter for IDE's based on Eclipse) could use. There were so much same or similar functionality.

To inherit GnuArmEclipse is of course another option, and maybe simpler. I could give it a try.